lib/frameworks/mocha.js

```diff
@@ -46,7 +47,7 @@ module.exports.run = function(config, specs, capabilities) {
     /**
      * enable generators if supported
      */
-    if(hasES6Support) {
+    if(hasES6Support && interfaces[config.mochaOpts.ui]) {
```

Comment:
Mocha has 5 interfaces. https://mochajs.org/#interfaces One of my projects uses one that you (without this PR) don't support.